From 8ee4c65eee2f02dccbdace6478d15fb3a6c7f9e0 Mon Sep 17 00:00:00 2001
From: Sujeet Baranwal <sbaranwal@marvell.com>
Date: Tue, 3 Dec 2019 13:09:31 -0800
Subject: [PATCH 0854/1921] proc: publish no of macs in octeontx_info node

no of macs is being read and published upon invocation
of /proc/octeontx_info.

Change-Id: I0da8112537809f23236165348dbba21ab085acfa
Signed-off-by: Sujeet Baranwal <sbaranwal@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/19761
Reviewed-by: Sujeet Kumar Baranwal <Sujeet.Baranwal@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/32745
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/soc/marvell/octeontx_info.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/soc/marvell/octeontx_info.c b/drivers/soc/marvell/octeontx_info.c
index 4a40efe58e65..a951b0dd825d 100644
--- a/drivers/soc/marvell/octeontx_info.c
+++ b/drivers/soc/marvell/octeontx_info.c
@@ -29,6 +29,7 @@ struct octtx_brd_info {
 	const char *board_revision;
 	const char *board_serial;
 	const char *board_model;
+	const char *board_num_of_mac;
 	int  dev_tree_parsed;
 	struct octeontx_info_mac_addr mac_addr;
 };
@@ -48,6 +49,7 @@ static int oct_brd_proc_show(struct seq_file *seq, void *v)
 	seq_printf(seq, "board_revision: %s\n", brd.board_revision);
 	seq_printf(seq, "board_serial_number: %s\n", brd.board_serial);
 	seq_printf(seq, "mac_addr_base: %pM\n", brd.mac_addr.s.bytes);
+	seq_printf(seq, "mac_addr_count: %s\n", brd.board_num_of_mac);
 
 	return  0;
 }
@@ -110,6 +112,14 @@ static int octtx_info_init(void)
 				brd.mac_addr.num = be64_to_cpu(mac_addr.num);
 		}
 
+
+		ret = of_property_read_string(np, "BOARD-MAC-ADDRESS-NUM",
+							&brd.board_num_of_mac);
+		if (ret) {
+			pr_warn("Board mac address number not available\n");
+			brd.board_num_of_mac = null_string;
+		}
+
 		brd.dev_tree_parsed = 1;
 	}
 
-- 
2.31.1

